Modules/@babylonjs/react-native/EngineView.tsx

```diff
@@ -90,10 +98,52 @@ export const EngineView: FunctionComponent<EngineViewProps> = (props: EngineView
         setFps(undefined);
     }, [props.camera, props.displayFrameRate]);
 
+    // Call onInitialized if provided, and include the callback for takeSnapshot.
+    useEffect(() => {
+        if (props.onInitialized) {
+            props.onInitialized({
+                takeSnapshot: () => {
+                    if (!snapshotPromise.current) {
+                        let resolutionFunctions: { resolve: (data: string) => void, reject: () => void } | undefined;
+                        const promise = new Promise<string>((resolutionFunc, rejectionFunc) => {
+                            resolutionFunctions = {resolve: resolutionFunc, reject: rejectionFunc};
+                        });
+
+                        // Resolution functions should always be initialized.
+                        if (resolutionFunctions) {
+                            snapshotPromise.current = { promise: promise, resolve: resolutionFunctions.resolve, reject: resolutionFunctions.reject };
+                        }
+                        else {
+                            throw "Resolution functions not initialized after snapshot promise creation.";
+                        }
+
+                        UIManager.dispatchViewManagerCommand(
+                            findNodeHandle(engineViewRef.current),
+                            "takeSnapshot",
+                            []);
+
+                        return promise;
+                    }
+
+                    return snapshotPromise.promise;
```

Comment:
Isn't this broken? Doesn't it need to be `snapshotPromise.current.promise`?

Reply:
Yes good catch, fixed.  I guess it just returned undefined so it didn't actually error out, but definitely incorrect.